paasta_tools/frameworks/native_scheduler.py

```diff
@@ -125,22 +139,34 @@ def launch_tasks_for_offers(self, driver, offers):
         launched_tasks = []
 
         for offer in offers:
-            tasks = self.tasks_for_offer(driver, offer)
-            if tasks:
-                for task in tasks:
-                    self.tasks_with_flags.setdefault(
-                        task.task_id.value,
-                        MesosTaskParameters(
-                            health=None, mesos_task_state=TASK_STAGING))
-
-                operation = mesos_pb2.Offer.Operation()
-                operation.type = mesos_pb2.Offer.Operation.LAUNCH
-                operation.launch.task_infos.extend(tasks)
-                driver.acceptOffers([offer.id], [operation])
-
-                launched_tasks.extend(tasks)
-            else:
-                driver.declineOffer(offer.id)
+            with self.constraint_state_lock:
```

Comment:
i don't know threading properties of mesos driver, if we're 100% sure no multiple statusUpdate/resourceOffer calls can be invoked in parallel we can drop the lock, for now it's a case of "better safe than sorry". let me know if you're certain we can remove the lock.